File: Dagger2_Test/app/src/main/java/com/miraclehwan/dagger2_test/MainActivity.java
package com.miraclehwan.dagger2_test;

import android.os.Bundle;
import android.util.Log;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import com.miraclehwan.dagger2_test.di.DaggerMyComponent;
import com.miraclehwan.dagger2_test.di.MyComponent;
import com.miraclehwan.dagger2_test.model.Kitchen;

import javax.inject.Inject;

public class MainActivity extends AppCompatActivity {

    @Inject
    Kitchen mKitchen;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        MyComponent myComponent = DaggerMyComponent.create();
        myComponent.inject(this);

        isOrder();
    }

    private void isOrder() {
        boolean isOrder = mKitchen.isOrder();
        if (isOrder) {
            Log.e("isOrder", "order successful");
        } else {
            Log.e("isOrder", "order failed");
        }
    }
}

File: MiraclePub/app/src/main/java/com/miraclehwan/miraclepub/MainActivity.kt
package com.miraclehwan.miraclepub

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import com.miraclehwan.miraclepub.di.component.DaggerPubComponent
import com.miraclehwan.miraclepub.model.Server
import com.miraclehwan.miraclepub.model.ale.PaleAle
import com.miraclehwan.miraclepub.model.ale.Porter
import com.miraclehwan.miraclepub.model.ale.Stout
import com.miraclehwan.miraclepub.model.larger.Pilsner
import javax.inject.Inject
import javax.inject.Provider

class MainActivity : AppCompatActivity() {

    @Inject
    lateinit var mServerProvider: Provider<Server>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val pubCompoent = DaggerPubComponent.create()
        pubCompoent.inject(this)

        val server1 = mServerProvider.get()
        val server2 = mServerProvider.get()

        Log.e("test", server1.mName)
        Log.e("test", server2.mName)

        val pilsner = server1.makeBeer(Pilsner::class.java)
        Log.e("test", pilsner.toString())
        val pilsner2 = server1.makeBeer(Pilsner::class.java)
        Log.e("test", pilsner2.toString())
    }
}
